# Model Selection

Logistic Regression

In [88]:
from sklearn.linear_model import LogisticRegression

model = LogisticRegression()
model.fit(X_train, y_train)

LogisticRegression()

In [89]:
y_pred = model.predict(X_test)

# Model evaluation

In [90]:
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_test, y_pred)

In [91]:
from sklearn.metrics import accuracy_score, precision_score, recall_score

y_pred = model.predict(X_test)

accuracy = accuracy_score(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)

print("Accuracy:", accuracy)
print("Precision:", precision)
print("Recall:", recall)

Accuracy: 0.7974443918599148
Precision: 0.6247464503042597
Recall: 0.558983666061706


The accuracy of predicting churn using logistic regression looks quite high (79.74%)

In [92]:
from sklearn.ensemble import AdaBoostClassifier
from sklearn.metrics import accuracy_score  

ada = AdaBoostClassifier()
ada.fit(X_train, y_train)
ada_preds = ada.predict(X_test)

accuracy = accuracy_score(y_test, ada_preds)  

print("Accuracy:", accuracy)

Accuracy: 0.795551348793185


Performing hyperparameter tuning using grid search

In [93]:
from sklearn.model_selection import GridSearchCV, StratifiedShuffleSplit
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import AdaBoostClassifier

n_estimators = [100, 140, 145, 150, 160, 170, 175, 180, 185]
cv = StratifiedShuffleSplit(n_splits=10, test_size=0.30, random_state=15)
learning_r = [0.1, 1, 0.01, 0.5]

parameters = {
    'base_estimator': [LogisticRegression()],  
    'n_estimators': n_estimators,
    'learning_rate': learning_r
}

grid = GridSearchCV(
    AdaBoostClassifier(),
    param_grid=parameters,
    cv=cv,
    n_jobs=-1
)

grid.fit(X_train, y_train)

GridSearchCV(cv=StratifiedShuffleSplit(n_splits=10, random_state=15, test_size=0.3,
            train_size=None),
             estimator=AdaBoostClassifier(), n_jobs=-1,
             param_grid={'base_estimator': [LogisticRegression()],
                         'learning_rate': [0.1, 1, 0.01, 0.5],
                         'n_estimators': [100, 140, 145, 150, 160, 170, 175,
                                          180, 185]})

In [94]:
adpred=grid.predict(X_test)
from sklearn import metrics
metrics.accuracy_score(y_test,adpred)

0.7941315664931378